### ISSUE TITLE
Allow constructors to optionally be able to move whilst operating

### ISSUE THREAD

**Issue by TheSilverHornet** [beyond-all-reason/spring#1600]
Specifically, this is needed to allow the Printer (corvac) to exist as a single unit, rather than two grafted together, which has multiple problems.

Use case is to allow the unit to do drive-by repairs and reclaims; having to halt is highly hazardous for the health of frontline units.

**Comment by sprunk** [comment]
Thoughts:
* this requires decoupling the concept of "buildpower target" from the current command
* good opportunity to introduce multiple buildpower streams per unit (so you can e.g. repair up to 3 things at once without having to graft extra units as well)
* units can already have multiple weapons, weapons already have individual targets, there is already extensive Lua API for weapons, and weapons already have a lot of relevant traits like range. Perhaps add a Buildpower "weapon" type?

**Comment by TheSilverHornet** [comment]
I should maybe also add that this is blocking a unit needed for balance reasons from getting into the game. As I understand it, GDT dislike the current hybrid unit implementation, and therefore object to the unit being formally used until the engine supports it properly.

**Comment by WatchTheFort** [comment]
There are a bunch of implementation questions we'll have to answer.

- What happens when the distance between them grows greater than the constructor's build distance?
- Does this apply only to repair? Reclaim? Resurrect? Capture? Construction?

**Comment by sprunk** [comment]
With weapons:
1) same as for weapons: just stops "shooting"; chases if there is an active command and not just set-target.
2) the "canCapture" etc tags would be moved to the weapon so you can control it accordingly. If you just want (say) repair-on-the-move but not capture then you split it into two weapons, slaved to each other, and make only one work on the move.

**Comment by WatchTheFort** [comment]
>     2. the "canCapture" etc tags would be moved to the weapon so you can control it accordingly. If you just want (say) repair-on-the-move but not capture then you split it into two weapons, slaved to each other, and make only one work on the move.

That would mean adding one weapon per activity (build, repair, reclaim, etc.), unless each unit would just have two weapons, one that works on the move and one that doesn't, and assign each activity to one of the weapons?

**Comment by sprunk** [comment]
Yes, just two, and only those that have some types of work usable on the move and not others. If every type of work is usable (or not) on the move then you'd just have 1. Ditto for any combination of different traits you may want (e.g. you can repair from a range of 987, but capture from a range of 123).
